src/Behat/Behat/Context/Reader/AttributedContextReader.php

```diff
@@ -0,0 +1,70 @@
+<?php
+
+/*
+ * This file is part of the Behat.
+ * (c) Konstantin Kudryashov <ever.zet@gmail.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Behat\Behat\Context\Reader;
+
+use Behat\Behat\Context\Attribute\AttributeReader;
+use Behat\Behat\Context\Environment\ContextEnvironment;
+use ReflectionClass;
+use ReflectionMethod;
+
+/**
+ * Reads context callees by Attributes using registered Attribute readers.
+ *
+ * @author Konstantin Kudryashov <ever.zet@gmail.com>
+ */
+class AttributedContextReader implements ContextReader
```

Comment:
Naming things ...

I don't really like `AttributedContextReader`, but I don't have any better ideas either.
Pro is that it's discoverable wrt `AnnotationContextReader` so it kind of makes sense.

Reply:
`AttributeContextReader` seems better to me